upgrade/upgrade-scripts/common.sh

```diff
@@ -193,6 +193,17 @@ function copy_dataset_property() {
 		die "failed to set property '$PROP_NAME=$PROP_VALUE' for '$DST_DATASET'"
 }
 
+function copy_dataset_property_to_snapshots_recursive() {
+	local PROP_NAME="$1"
+	local SRC_DATASET="$2"
+	local SNAPSHOT_NAME="$3"
+
+	for dir in $(zfs list -rH -o name "$SRC_DATASET"); do
```

Comment:
I'd prefer `dataset` (or `ds` for short) to `dir`, since it's not really a "directory" (which is what I think of when I see "dir"); but it doesn't really matter so this is fine too.